src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverter.cs

```diff
@@ -35,10 +35,18 @@ protected override bool SetDocumentFields<TBuilder>(LazyStringValue key, LazyStr
             bool innerShouldSkip = false;
             if (indexField.Vector != null)
             {
-                if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, ((AutoVectorOptions)indexField.Vector).SourceFieldName, out value) == false)
+                var autoVectorOptions = (AutoVectorOptions)indexField.Vector;
+                
+                if (BlittableJsonTraverserHelper.TryRead(_blittableTraverser, document, autoVectorOptions.SourceFieldName, out value) == false)
                     continue;
-                var vector = AbstractStaticIndexBase.CreateVector(indexField, value, isAutoIndex: true);
-                InsertRegularField(indexField, vector, indexContext, builder, sourceDocument,  out innerShouldSkip);
+
+                object vector;
+
+                if (autoVectorOptions.EmbeddingsGenerationTaskIdentifier != null)
```

Comment:
I don't follow why I care if the vector has a task id?

Reply:
,When a vector has a task id, it means we need to read it from the attachment etc. Otherwise, we can assume that the vector is in the document.